### PR TITLE
tests: Show summary of image sizes

### LINKED ISSUE #162: show summary of image sizes at end of tests
It would be useful to see a formatted list of the images and initrds at the end of the test, showing the sizes of each.

### PR DESCRIPTION
**Pull request by jodh-intel**
At the end of the tests, show a summary of the sizes of all the images
and initrds along with their sizes.

Fixes #162.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Added @marcov's idea to show sizes in MB (too), so now we get a nice size-sorted summary at the end of the test run:

```
INFO: Image sizes (bytes) and types (small [better] to large)

       Bytes	      MB	Type    	Name                

     9269859	    8.84	initrd  	alpine              
   134217728	  128.00	image   	clearlinux          
   268435456	  256.00	image   	fedora              
   268435456	  256.00	image   	ubuntu              
   536870912	  512.00	image   	centos              
   536870912	  512.00	image   	euleros             
```
.. and check out those size differences! :smile:

**Comment by jodh-intel** [comment]
I've added sizes for the rootfs's too now. We can finesse in future PRs I think ;)

**Comment by jodh-intel** [comment]
Here's what we have now:

```
INFO: Image and rootfs sizes (in bytes and MB), smallest image first:

 image-bytes	  image-MB	rootfs-bytes	 rootfs-MB	Type    	Name
     9271554	      8.84	    28279412	     26.96	initrd  	alpine
   134217728	    128.00	    69513295	     66.29	image   	clearlinux
   268435456	    256.00	   140046086	    133.55	image   	fedora
   268435456	    256.00	   142485451	    135.88	image   	ubuntu
   536870912	    512.00	   343398171	    327.49	image   	euleros
   536870912	    512.00	   352602911	    336.26	image   	centos
```

File: .ci/setup.sh [hunk 1]
```diff
@@ -14,13 +14,13 @@ source "${cidir}/lib.sh"
 source /etc/os-release
 
 if [ "$ID" == fedora ];then
-	sudo -E dnf -y install automake yamllint coreutils moreutils
+	sudo -E dnf -y install automake yamllint coreutils moreutils bc
```
Comment (grahamwhaley) [inline .ci/setup.sh]: heh heh - we could factor out a bunch of this to 'distro-common' packages by the look of it. I know how much you hate duplication @jodh-intel ;-)

File: tests/test_images.sh [hunk 2]
```diff
@@ -270,6 +343,8 @@ handle_options()
 
 		build_rootfs "${distro}" "${rootfs}"
 
+		local rootfs_size=$(du -sb "${rootfs}" | awk '{print $1}')
```
Comment (grahamwhaley) [inline tests/test_images.sh]: Ah, I think I get it - you are taking the `du` of the rootfs dir on the build machine itself. That's probably good enough. I was of course imagining we'd have to go look inside the image file itself to get the actual size...
Side question for @jcvenegas - we don't strip the files during the conversion from the rootfs dir to the image file itself, do we?

**Comment by jodh-intel** [comment]
Ping @kata-containers/builder!

**Comment by jodh-intel** [comment]
... Ping? :)

File: tests/test_images.sh (hunk 2 above)
Comment (jcvenegas) [reply tests/test_images.sh]: @grahamwhaley  we dont that  may be a good think to do, lets open an issue for that.

**Comment by jcvenegas** [comment]
lgtm

**Pull request merged**